alchemy/src/aws/control/resource.ts

```diff
@@ -0,0 +1,230 @@
+ * known at compile time or when not using the typed Proxy interface.
+ *
+ * For the strongly-typed Proxy interface (e.g., `AWS.S3.Bucket(...)`), Alchemy
+ * uses internal handlers generated by the `createResourceType` factory function.
+ *
+ * Creates and manages AWS resources using the Cloud Control API.
+ *
+ * @example
+ * // Create an S3 bucket
+ * const bucket = await CloudControlResource("my-bucket", {
+ *   typeName: "AWS::S3::Bucket",
+ *   desiredState: {
+ *     BucketName: "my-unique-bucket-name",
+ *     VersioningConfiguration: {
+ *       Status: "Enabled"
+ *     }
+ *   }
+ * });
+ *
+ * @example
+ * // Create a DynamoDB table
+ * const table = await CloudControlResource("users-table", {
+ *   typeName: "AWS::DynamoDB::Table",
+ *   desiredState: {
+ *     TableName: "users",
+ *     AttributeDefinitions: [
+ *       {
+ *         AttributeName: "id",
+ *         AttributeType: "S"
+ *       }
+ *     ],
+ *     KeySchema: [
+ *       {
+ *         AttributeName: "id",
+ *         KeyType: "HASH"
+ *       }
+ *     ],
+ *     ProvisionedThroughput: {
+ *       ReadCapacityUnits: 5,
+ *       WriteCapacityUnits: 5
+ *     }
+ *   }
+ * });
+ */
+export const CloudControlResource = Resource(
+  "aws::CloudControlResource",
+  CloudControlLifecycle,
+);
+
+async function CloudControlLifecycle(
+  this: Context<CloudControlResource, CloudControlResourceProps>,
+  id: string,
+  props: CloudControlResourceProps,
+) {
+  const client = await createCloudControlClient({
+    region: props.region,
+    accessKeyId: props.accessKeyId,
+    secretAccessKey: props.secretAccessKey,
+    sessionToken: props.sessionToken,
+  });
+
+  if (this.phase === "delete") {
+    if (this.output?.id) {
+      try {
+        await client.deleteResource(props.typeName, this.output.id);
+      } catch (error) {
+        // Log but don't throw on cleanup errors
+        console.error(`Error deleting resource ${id}:`, error);
+      }
+    }
+    return this.destroy();
+  }
+
+  let response: ProgressEvent | undefined;
+  if (this.phase === "update" && this.output?.id) {
+    // Update existing resource
+    response = await client.updateResource(
+      props.typeName,
+      this.output.id,
+      compare(this.output.desiredState, props.desiredState),
+    );
+  } else {
+    // Create new resource
+    try {
+      response = await client.createResource(
+        props.typeName,
+        props.desiredState,
+      );
+    } catch (error) {
+      if (error instanceof AlreadyExistsError) {
+        console.log("Resource already exists, updating", error.progressEvent);
+
+        const resource = await client.getResource(
+          props.typeName,
+          error.progressEvent.Identifier!,
+        );
+
+        response = await client.updateResource(
+          props.typeName,
+          error.progressEvent.Identifier!,
+          // TODO(sam): some of these properties may be "output" properties
```

Comment:
@sam-goodwin not sure about this one actually.
Picking only the properties from the desired state wouldn't work right?